internal/junit/junit.go

```diff
@@ -72,3 +75,42 @@ func Parse(data []byte) (TestSuites, error) {
 
 	return tss, err
 }
+
+// GetFailedTests get failed test list from the JUnit report.
+func GetFailedTests(report TestSuites) []string {
+	classes := map[string]bool{}
+	for _, s := range report.TestSuites {
+		for _, tc := range s.TestCases {
+			if tc.Error != "" || tc.Failure != "" {
+				if tc.Name != "" {
+					classes[fmt.Sprintf("%s/%s", tc.ClassName, tc.Name)] = true
+				} else {
+					classes[tc.ClassName] = true
+				}
+			}
+		}
+	}
+	return getKeysFromMap(classes)
+}
+
+// GetFailedClasses get failed class list from the JUnit report.
+func GetFailedClasses(report TestSuites) []string {
+	classes := map[string]bool{}
+
+	for _, s := range report.TestSuites {
+		for _, tc := range s.TestCases {
+			if tc.Error != "" || tc.Failure != "" {
+				classes[tc.ClassName] = true
+			}
+		}
+	}
+	return getKeysFromMap(classes)
+}
```

Comment:
I don't think anyone of us will know what the difference is between these two in a few weeks.

1. Both seem to be capable of returning a list of classes. Which do I choose?
2. `GetFailedTests` can return either a class or class+method combo, but doesn't explain anything about this in the docs.
3.`GetFailedTests` uses `tc.Name != ""` to decide on what to return, but as a reader, I can't tell why. And there's no inline explanation as to why this is the case. It'd be equally logical that `tc.Name` is a fully qualified name.

Reply:
> Both seem to be capable of returning a list of classes. Which do I choose?
If the use case is getting test list, use `getFailedTests`. If the use case is getting class list, use `getFailedClasses`.
`getFailedTests` is only having a fallback to return className. I'll add inline comments to explain it